### PR TITLE
AF-1424: Support SSH keystore with workbench

### PR DESCRIPTION
**Pull request by pefernan**
AF-1275: SSH Keystore backend
AF-1274: SSH Keystore UI
AF-1277: Integrate SSH Keystore with VFS SSH

@ederign @domhanak could you please take a look?

This is part of an ensemble, please merge with:
https://github.com/kiegroup/appformer/pull/492
https://github.com/kiegroup/kie-wb-common/pull/2177
https://github.com/kiegroup/kie-wb-distributions/pull/829
https://github.com/kiegroup/drools-wb/pull/959
https://github.com/kiegroup/jbpm-wb/pull/1215
https://github.com/kiegroup/optaplanner-wb/pull/308
https://github.com/kiegroup/kie-docs/pull/1156

### CODE REVIEW

**Comment by domhanak** [comment]
@sutaakar Hello! Could you please do a review from Cloud point of view? Do you see anything that may not work as expected on openshift?

**Comment by pefernan** [comment]
@domhanak @sutaakar notice that this is a default SSH KeyStore implementation. At some point an integration with openshift should be delivered.

**Comment by domhanak** [comment]
@pefernan Great! Thanks for the heads up I missed that info :) I met with Karel today and he pinpointed 2 areas that might require change or special attention with Openshift integration.

- Keystores should be stored on managed persistent volume, and we need to take care of how it is created or handled
- We need to be extra careful how we handle concurrent access to the keystore on openshift

_I am continuing with local testing_

**Comment by pefernan** [comment]
please jenkins execute full downstream build

**Comment by domhanak** [comment]
please jenkins execute full downstream build

**Comment by pefernan** [comment]
please jenkins execute full downstream build

**Comment by domhanak** [comment]
please jenkins execute full downstream build

**Comment by domhanak** [comment]
please jenkins execute full downstream build

**Comment by domhanak** [comment]
please jenkins execute full downstream build

**Comment by pefernan** [comment]
please jenkins execute full downstream build

**Comment by domhanak** [comment]
Roses are red,
violets are blue,
Jenkins execute full downstream build!

**Comment by pefernan** [comment]
please jenkins execute full downstream build

**Comment by pefernan** [comment]
So @domhanak is it approved?

**Comment by ederign** [comment]
@pefernan could you please fix the conflicts ?

File: uberfire-ssh/uberfire-ssh-backend/src/main/java/org/uberfire/ssh/service/backend/auth/impl/SSHKeyAuthenticatorImpl.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    private final WorkbenchUserManager userManager;
+
+    @Inject
+    public SSHKeyAuthenticatorImpl(final SSHKeyStoreService keyStoreService, final Instance<WorkbenchUserManager> workbenchUserManagerInstance) {
+        this.keyStoreService = keyStoreService;
+
+        if (!workbenchUserManagerInstance.isUnsatisfied() && !workbenchUserManagerInstance.isAmbiguous()) {
+            this.userManager = workbenchUserManagerInstance.get();
+        } else {
+            LOGGER.warn("Cannot find any implmenentation of 'WorkbenchUserManager'. Loading default implemntation from SSH module");
```
[the remark this answers is not in the document]
Comment (pefernan) [reply uberfire-ssh/uberfire-ssh-backend/src/main/java/org/uberfire/ssh/service/backend/auth/impl/SSHKeyAuthenticatorImpl.java]: tow tyops in teh same lien.

**Comment by pefernan** [comment]
@ederign fixed!

**Comment by pefernan** [comment]
please jenkins execute full downstream build

**Comment by domhanak** [comment]
You can do it Jenkins, just believe in yourself.

Let's make sure there are only random failures.

jenkins execute full downstream build

**Comment by domhanak** [comment]
@pefernan I just realized we only tested this with old wildfly. I am going to retest it in the morning, but If you can do it sooner feel free to do so :)

**Comment by domhanak** [comment]
@jhrcek Any ideas about the failures? I see lots of ISPN failures

**Comment by ederign** [comment]
@domhanak @jhrcek the ISPN issue is fixed here: https://github.com/kiegroup/droolsjbpm-integration/pull/1617#pullrequestreview-170431368 

One thing that we need to figure out is why this was not caught by jenkins build.

**Pull request merged**